services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py

```diff
@@ -286,30 +315,65 @@ async def task_save_state(
     mounted_volumes: MountedVolumes,
     app: FastAPI,
 ) -> None:
+    """
+    Saves the states of the service.
+    If a legacy archive is detected, it will be removed after
+    saving the new format.
+    """
+
+    async def _push_and_remove_legacy_archive(
+        state_path: Path, root_progress: ProgressBarData
+    ) -> None:
+        project_id_str = f"{settings.DY_SIDECAR_PROJECT_ID}"
+        node_uuid_str = f"{settings.DY_SIDECAR_NODE_ID}"
+
+        await data_manager.push_directory(
+            user_id=settings.DY_SIDECAR_USER_ID,
+            project_id=project_id_str,
+            node_uuid=node_uuid_str,
+            source_path=state_path,
+            # NOTE: interface was broken on purpose, next PR wil fix it
+            # and will cleanup how rclone is used
+            r_clone_settings=settings.DY_SIDECAR_R_CLONE_SETTINGS,
+            exclude_patterns=mounted_volumes.state_exclude,
+            io_log_redirect_cb=functools.partial(
+                post_sidecar_log_message, app, log_level=logging.INFO
+            ),
+            progress_bar=root_progress,
+        )
+        archive_exists = await data_manager.state_metadata_entry_exists(
+            user_id=settings.DY_SIDECAR_USER_ID,
+            project_id=project_id_str,
+            node_uuid=node_uuid_str,
+            path=state_path,
+            is_archive=True,
+        )
+
+        if not archive_exists:
+            return
+
+        with log_context(_logger, logging.INFO, "removing legacy data archive"):
+            await data_manager.delete_legacy_archive(
+                user_id=settings.DY_SIDECAR_USER_ID,
+                project_id=project_id_str,
+                node_uuid=node_uuid_str,
+                path=state_path,
+            )
+
     progress.update(message="starting state save", percent=0.0)
+    state_paths = [  # pylint: disable=unnecessary-comprehension # noqa: C416
+        x for x in mounted_volumes.disk_state_paths_iter()
```

Comment:
```python
state_paths = list(mounted_volumes.disk_state_paths_iter()) 
```
btw  do you really need a iter for 2 - 3 elements?

Reply:
Some design decision, which I will not remove now. Maybe in the future.